### PR TITLE
Vatasha/ally 639/migrate dd alert channel api v2 fix

### PR DESCRIPTION
**Pull request by vatasha**
## Summary

We're currently migrating alert channels to v2 to be able to support alert channels at the org level. 

## How did you test this change?
There are updated tests

## Issue

https://lacework.atlassian.net/browse/ALLY-639?atlOrigin=eyJpIjoiMjM4NmRkMzFmNzNlNDMwMGFhYmM2N2YwZjg0ZjFkYWYiLCJwIjoiaiJ9

### CODE REVIEW

**Review by afiune — APPROVED** [review]
Please, squash and merge this PR with a good message.

File: api/alert_channel_datadog.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+// UpdateDatadog updates a single instance of a Datadog integration on the Lacework server
+func (svc *AlertChannelsService) UpdateDatadog(data AlertChannel) (response DatadogAlertChannelResponseV2, err error) {
+	err = svc.update(data.ID(), data, &response)
+	return
+}
+
+type DatadogDataV2 struct {
+	APIKey      string `json:"apiKey"`
+	DatadogSite string `json:"datadogSite,omitempty"`
+	DatadogType string `json:"datadogType,omitempty"`
```
Comment (dmurray-lacework) [inline api/alert_channel_datadog.go:37]: Is omitempty needed here? Schema says they are required fields

File: api/alert_channel_datadog.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+// UpdateDatadog updates a single instance of a Datadog integration on the Lacework server
+func (svc *AlertChannelsService) UpdateDatadog(data AlertChannel) (response DatadogAlertChannelResponseV2, err error) {
+	err = svc.update(data.ID(), data, &response)
+	return
+}
+
+type DatadogDataV2 struct {
+	APIKey      string `json:"apiKey"`
```
Comment (dmurray-lacework) [inline api/alert_channel_datadog.go:35]: Elsewhere in the go-sdk do we use "API" OR "Api"?

**Review by dmurray-lacework — CHANGES_REQUESTED** [review]
We should also include the CLI Create integration changes in these migrations?

cli/cmd/integration_datadog.go:83
cli/cmd/integration_datadog.go:92

File: api/alert_channel_datadog.go (hunk 1 above)
Comment (vatasha) [reply api/alert_channel_datadog.go:37]: There is a default value which is why I have omitempty

**Pull request merged**